pkg/endpoints/routes_test.go

```diff
@@ -1,387 +1,523 @@
 		}
-		if resource == nil {
-			return nil
+		// "Content-Location" header should be of the form /some/path/<resource>/name
+		postedResourceName := contentLocationSlice[0][strings.LastIndex(contentLocationSlice[0], "/")+1:]
+		t.Log("Posted resoure name:", postedResourceName)
+		routeNameMap[route] = postedResourceName
+		httpReq := dummyHTTPRequest("GET", fmt.Sprintf("%s%s", server.URL, contentLocationSlice[0]), httpRequestBody)
+		_, err := http.DefaultClient.Do(httpReq)
+		if err != nil {
+			t.Fatalf("Unable to locate resource as specified by 'Content-Location' header: %s", contentLocationSlice[0])
 		}
-		var requestBody *bytes.Reader
-		var body []byte
-		var err error
+	}
+}
 
-		if resourceType == "pipelineruns" {
-			// request body should be status: PipelineRunCancelled
-			if httpMethod == http.MethodPut {
-				updateBody := PipelineRunUpdateBody{}
-				updateBody.STATUS = "PipelineRunCancelled"
-				body, err = json.Marshal(updateBody)
-				if err != nil {
-					t.Errorf("Failed to marshal resource type for PipelineRunUpdate: %s", err)
-					return nil
+// ALL and ONLY resourceTypes with POST/PUT routes must implement
+// Returns stub to be marshalled for dashboard route validation
+func fakeStub(t *testing.T, r *Resource, httpMethod, resourceType, namespace, resourceName string) interface{} {
+	t.Logf("Getting fake resource type: %s\n", resourceType)
+	// Cases based on resource substring in mux routes (lowercase)
+	switch resourceType {
+	case "pipelinerun":
+		// ManualPipelineRun require the referenced pipeline to have the same name as the pipeline being created
+		switch httpMethod {
+		case "POST":
+			// Avoid creation collision when checking GET routes for stub
+			_, err := r.PipelineClient.TektonV1alpha1().Pipelines(namespace).Get(resourceName, metav1.GetOptions{})
+			// If pipeline does not exist
+			if err != nil {
+				// TODO: embed in pipelineRun once pipelineSpec is supported
+				// Create pipeline ref for ManualPipelineRun
+				pipeline := v1alpha1.Pipeline{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      resourceName,
+						Namespace: namespace,
+					},
+					Spec: v1alpha1.PipelineSpec{},
 				}
-			} else {
-				createBody := ManualPipelineRun{}
-				createBody.PIPELINENAME = "fakepipeline"
-				body, err = json.Marshal(createBody)
+				_, err := r.PipelineClient.TektonV1alpha1().Pipelines(namespace).Create(&pipeline)
 				if err != nil {
-					t.Errorf("Failed to marshal resource type for PipelineRunCreate: %s", err)
-					return nil
+					t.Fatalf("Error creating pipeline for pipelinerun: %v\n", err)
 				}
 			}
-
-		} else {
-			body, err = json.Marshal(&resource)
-			if err != nil {
-				t.Errorf("Failed to marshal resource type: %s, error: %s", resourceType, err)
-				return nil
+			return ManualPipelineRun{
+				PIPELINENAME: resourceName,
 			}
+		case "PUT":
+			return PipelineRunUpdateBody{
+				v1alpha1.PipelineRunSpecStatusCancelled,
```

Comment:
nit: To be explicit, can you please change this to `STATUS: v1alpha1.PipelineRunSpecStatusCancelled,`